### PR TITLE
Upgrade flow

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Spread on types doesn't quite work as expected see: https://github.com/facebook/flow/issues/3534#issuecomment-287580240

### CODE REVIEW

File: shared/actions/chat/index.js [hunk 1]
```diff
@@ -933,7 +933,7 @@ function _threadIsCleared(originalAction: Action, checkAction: Action): boolean
   return (
     originalAction.type === 'chat:loadMoreMessages' &&
     checkAction.type === 'chat:clearMessages' &&
-    originalAction.conversationIDKey === checkAction.conversationIDKey
+    originalAction.payload.conversationIDKey === checkAction.payload.conversationIDKey
```
Comment (MarcoPolo) [inline shared/actions/chat/index.js]: Didn't you fix this @chrisnojima ?

Comment (chrisnojima) [inline shared/actions/chat/index.js]: it wasn't merged

File: shared/login/signup/request-invite.js [hunk 2]
```diff
@@ -38,12 +50,16 @@ class RequestInvite extends Component {
...
-// $FlowIssue type this connector
 export default connect(
   state => ({
     emailErrorText: state.signup.emailError,
     nameErrorText: state.signup.nameError,
     waiting: state.signup.waiting,
   }),
-  dispatch => bindActionCreators(signupActions, dispatch)
+  (dispatch: Dispatch<*>) => ({
+    // $FlowIssue with thunks
```
Comment (chrisnojima) [inline shared/login/signup/request-invite.js]: what is the flow issue?

File: shared/package.json [hunk 3]
```diff
@@ -148,7 +147,7 @@
     "eslint-plugin-react": "7.0.1",
     "eslint-plugin-standard": "3.0.1",
     "file-loader": "0.11.1",
-    "flow-bin": "0.41.0",
+    "flow-bin": "^0.48.0",
```
Comment (chrisnojima) [inline shared/package.json]: likely remove the ^

File: shared/settings/notifications/index.js.flow [hunk 4]
```diff
@@ -22,8 +22,8 @@ export type Props = {
   },
   onRefresh: () => void,
   onSave: () => void,
-  onToggle: (name: string) => void,
-  onToggleUnsubscribeAll: () => void,
+  onToggle: (gorupName: string, name: string) => void,
```
Comment (chrisnojima) [inline shared/settings/notifications/index.js.flow]: typo

File: shared/login/signup/request-invite.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/login/signup/request-invite.js]: the normal `Dispatch` type from redux is looking for a function that returns an object, but thunks return another fn

**Pull request merged**